feat: add authentication feature

samthelightbender opened the pull request:
### feat: Add Login (Email and Password) and Logout Functionality

#### Description

This pull request introduces the initial phase of the authentication feature, focusing specifically on **login and logout** functionality. It lays the groundwork for the complete authentication system by setting up the necessary database models and a seeding script for test users.

#### Key Changes:

* Added `Session` and `Verification` models to the Prisma schema to support authentication.
* Created API endpoints for the login and logout processes.
* Implemented a seeding script to create a test user for login.
* Added npm scripts to streamline the database setup process.

#### Setup and Testing Instructions

To get this feature running locally, please follow these steps in order:

1.  **Add Environment Variables**
    Make sure you add the following to your `.env.development.example` file.

    ```env
    DATABASE_URL="postgresql://USER:PASSWORD@HOST:PORT/DATABASE"
    BETTER_AUTH_URL="http://your-app-url.com:port"
    BETTER_AUTH_SECRET="your-super-secret-key-here"
    ```
    > **Note:** The `BETTER_AUTH_SECRET` should be a securely generated secret.

2.  **Generate Prisma Client**
    Run this command to generate the Prisma Client according to the latest schema.
    ```bash
    npm run db:generate:dev
    ```

3.  **Run Database Migrations**
    This command will apply the new schema changes to your database.
    ```bash
    npm run db:migrate:dev
    ```

4.  **Seed the Database**
    Use this command to populate the database with initial user data for testing purposes.
    ```bash
    npm run db:seed:dev
    ```

After completing these steps, the application will be ready for testing the login and logout features.

samthelightbender committed:
feat: add authentication feature

samthelightbender committed:
migration: remove table that unrelated with authentication feature

zainphp reviewed (changes requested):
please fix error lint

> <img width="1083" height="443" alt="image" src="https://github.com/user-attachments/assets/27ecc3af-3f81-4c3f-ab8a-3e433ef8164a" />

samthelightbender committed:
lint: fix lint error

samthelightbender commented:
fixed lint error, please check again 🚀

samthelightbender committed:
Fix: Update Prisma schema path in npm scripts

Changed the Prisma schema path in all related npm scripts from 'src/prisma/schema.prisma' to 'prisma/schema.prisma' to reflect the new schema location.

samthelightbender committed:
Update .env.example

zainphp committed:
Refines Prisma configuration and updates linting scripts

Centralizes the Prisma schema path within `prisma.config.ts`, enabling the removal of redundant `--schema` arguments from database-related npm scripts.

Integrates `prisma format` into the `lint` and `lint:fix` commands to ensure consistent code formatting.

Removes the unused `eslint-plugin-naming` dependency and cleans up an unused `lucide-react` import and unutilized `ctx` parameters in authentication callbacks.

zainphp committed:
Prepares schema for initial authentication feature

Temporarily comments out models and fields unrelated to core authentication. This includes `Post`, `Like`, `Comment`, `Report`, `Follow`, and `LeaderboardSnapshot` models, along with their associated relations in the `User` model.

This change streamlines the database schema to focus solely on the initial authentication functionality, enabling a clearer development path. These features will be uncommented and implemented as they are developed.

Also renames the base migration file to accurately reflect its purpose in setting up the initial authentication tables.

zainphp committed:
Implements social media authentication options.

This commit adds support for authenticating with Google and GitHub in addition to the existing authentication mechanism. This enhancement allows users to login using their social media accounts, providing a more convenient and secure experience. The changes include adding new environment variables and registering the authentication routes.

zainphp committed:
remove unnecessary columns

zainphp committed:
fix lint error

zainphp reviewed (changes requested):
- add sign up form
- show error auth to frontend

samthelightbender committed:
Merge branch 'pemrogrammer:main' into auth

samthelightbender committed:
implement client login form error handling

samthelightbender committed:
registration and email verification using otp (without smtp)

samthelightbender committed:
update login: remove email param to email-verification

samthelightbender commented:
see the otp code at the verification table manually (for temporary)

samthelightbender committed:
add smtp support for OTP using Resend

samthelightbender committed:
add router dependency to fix lint

samthelightbender commented:
Read [Resend SMTP Docs](https://resend.com/docs/send-with-nextjs) and put the Resend API Key to your development env
`RESEND_API_KEY=`

zainphp committed:
Improves login form layout and workflow for improved user experience.

Enhances the accessibility and readability of the login form by providing clear labels and links for users.
Updates the layout to direct users to the correct authentication routes.
Adds social login buttons for GitHub.
Updates the CTAs on the landing page and header to direct users to the login/auth/register pages.
Improves overall user navigation and reduces confusion.
